library/pk_wrap.c

```diff
@@ -141,6 +141,82 @@ static int rsa_decrypt_wrap( void *ctx,
                 olen, input, output, osize ) );
 }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+static int rsa_encrypt_wrap( void *ctx,
+                    const unsigned char *input, size_t ilen,
+                    unsigned char *output, size_t *olen, size_t osize,
+                    int (*f_rng)(void *, unsigned char *, size_t), void *p_rng )
+{
+    mbedtls_rsa_context * rsa = (mbedtls_rsa_context *) ctx;
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    mbedtls_svc_key_id_t key_id = MBEDTLS_SVC_KEY_ID_INIT;
+    psa_status_t status;
+    mbedtls_pk_context key;
+    int key_len;
+    /* see RSA_PUR_DER_MAX_BYTES in pkwrite.c */
+    unsigned char buf[38 + 2 * MBEDTLS_MPI_MAX_SIZE];
+    mbedtls_pk_info_t pk_info = mbedtls_rsa_info;
+    psa_algorithm_t psa_alg_md;
+
+    ((void) f_rng);
+    ((void) p_rng);
+
+#if !defined(MBEDTLS_RSA_ALT)
+    switch( rsa->padding )
+    {
+        case MBEDTLS_RSA_PKCS_V15:
+            psa_alg_md = PSA_ALG_RSA_PKCS1V15_CRYPT;
+            break;
+
+        default:
+            return( MBEDTLS_ERR_RSA_INVALID_PADDING );
+    }
+#else
+    psa_alg_md = PSA_ALG_RSA_PKCS1V15_CRYPT;
+#endif
+
+    *olen = mbedtls_rsa_get_len( rsa );
+
+    if( *olen > osize )
+        return( MBEDTLS_ERR_RSA_OUTPUT_TOO_LARGE );
+
+    /* mbedtls_pk_write_pubkey() expects a full PK context;
+     * re-construct one to make it happy */
+    key.pk_info = &pk_info;
+    key.pk_ctx = ctx;
+    key_len = mbedtls_pk_write_pubkey_der( &key, buf, sizeof( buf ) );
+    if( key_len <= 0 )
+        return( MBEDTLS_ERR_PK_BAD_INPUT_DATA );
+
+    psa_set_key_usage_flags( &attributes, PSA_KEY_USAGE_ENCRYPT );
+    psa_set_key_algorithm( &attributes, psa_alg_md );
+    psa_set_key_type( &attributes, PSA_KEY_TYPE_RSA_PUBLIC_KEY );
+
+    status = psa_import_key( &attributes,
+                             buf + sizeof( buf ) - key_len, key_len,
+                             &key_id );
+    if( status != PSA_SUCCESS )
+    {
+        ret = mbedtls_psa_err_translate_pk( status );
+        goto cleanup;
+    }
+
+    status = psa_asymmetric_encrypt( key_id, psa_alg_md, input, ilen,
+                                     NULL, 0, output, osize, olen);
+    if( status != PSA_SUCCESS )
+    {
+        ret = mbedtls_psa_err_translate_pk( status );
+        goto cleanup;
+    }
+
+    ret = 0;
+
+cleanup:
+    psa_destroy_key( key_id );
```

Comment:
Status of `psa_destroy_key()` is ignored. 
`psa_destroy_key()` can only fail because operational errors (e.g. storage failure) which is very unlikely. But it is a good practice to handle these errors, so application can handle it.

Reply:
Excellent point. There's the same issue in the existing `ecdsa_verify_wrap()`. @superna9999 could you fix it too, while at it? (Technically it's out of scope for this PR, but it's probably easier to fix it here than to open another PR just for that.)

Reply:
OK, I have a doubt.
Which error should I return, the potential previous error or the `psa_destroy_key()` if such error occur ?

Reply:
I wrote the following logic :
```
    cleanup:
    status = psa_destroy_key( key_id );
    if( ret == 0 && status != PSA_SUCCESS )
        ret = mbedtls_psa_err_translate_pk( status );

    return( ret );
}
```
```

Reply:
There's also one place, added recently, in `pk_verify_ext()` that checks the status from `destroy` and returns either the original error if there was one before destroy, of the status from destroy. Whether it's better to return the original error or that form destroy may depend on the context, but if possible I think it would be good if all code (at least within each module) used the same idiom for that for handling destroy status.

Reply:
OK thanks

in `ssl_cli.c`:
```
         status = psa_raw_key_agreement(...)
         if( status != PSA_SUCCESS )                                                                 
             return( MBEDTLS_ERR_SSL_HW_ACCEL_FAILED );                                              
                                                                                                     
         status = psa_destroy_key( handshake->ecdh_psa_privkey );                                    
         if( status != PSA_SUCCESS )                                                                 
             return( MBEDTLS_ERR_SSL_HW_ACCEL_FAILED );    
```

I just discovered here `psa_destroy_key()` isn't called if `psa_raw_key_agreement()` fails.

in `pk_verify_ext()`:
```
         status = psa_verify_hash(...);
         destruction_status = psa_destroy_key( key_id );                                             
                                                                                                     
         if( status == PSA_SUCCESS && sig_len > mbedtls_pk_get_len( ctx ) )                          
             return( MBEDTLS_ERR_PK_SIG_LEN_MISMATCH );                                              
                                                                                                     
         if( status == PSA_ERROR_INVALID_SIGNATURE )                                                 
             return( MBEDTLS_ERR_RSA_VERIFY_FAILED );                                                
                                                                                                     
         if( status == PSA_SUCCESS )                                                                 
             status = destruction_status;                                                            
                                                                                                     
         return( mbedtls_psa_err_translate_pk( status ) );    
```

Both are prioritizing the main error, and return the `psa_destroy_key()` if the previous didn't fail.
I think it matches the logic I wrote earlier.